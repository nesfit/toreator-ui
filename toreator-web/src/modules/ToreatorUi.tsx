import * as React from "react";
import {IntlProvider} from "react-intl";
import {Provider} from "react-redux";
import {BrowserRouter as Router, Switch} from "react-router-dom";
import {ThemeProvider} from "styled-components";
import store from "../common/store/store";
import DynamicRoute from "./shared/components/Router";
import theme from "./shared/styles/theme";

// @ts-ignore
const lazy = React.lazy;

export const routes = {
  homePage: () =>
    // @ts-ignore
    lazy(() => import(/* webpackChunkName: "HomePage" */ "./home-page")),
};
interface Props {
  defaultLocale: string;
  locale: string;
  messages: object;
}

class ToreatorUi extends React.PureComponent<Props> {
  public static defaultProps = {
    defaultLocale: "en",
    locale: "en",
    messages: {},
  };

  public render() {
    const {defaultLocale, locale, messages} = this.props;

    return (
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <IntlProvider
            defaultLocale={defaultLocale}
            locale={locale}
            messages={messages}
          >
            <Router>
              <Switch>
                <DynamicRoute
                  exact={true}
                  path="/"
                  loadAsyncModuleFn={routes.homePage}
                />
              </Switch>
            </Router>
          </IntlProvider>
        </ThemeProvider>
      </Provider>
    );
  }
}

export default ToreatorUi;
