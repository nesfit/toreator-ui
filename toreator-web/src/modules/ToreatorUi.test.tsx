import {shallow} from "enzyme";
import * as React from "react";
import ToreatorUi, {routes} from "./ToreatorUi";

describe("<ToreatorUi/>", () => {
  test("default", () => {
    const wrapper = shallow(<ToreatorUi />);
    expect(wrapper).toMatchInlineSnapshot(`
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Array [
          "36em",
          "48em",
          "62em",
          "75em",
        ],
        "fontSizes": Array [],
        "lineHeights": Array [],
        "media": Object {
          "lg": [Function],
          "md": [Function],
          "sm": [Function],
          "xl": [Function],
        },
        "palette": Object {
          "greenColor": "#20b04d",
          "pinkColor": "#d84f87",
          "pinkHoverColor": "#d84f97",
          "purpleColor": "#52489C",
          "redColor": "#e40421",
        },
        "space": Array [
          0,
          4,
          8,
          12,
          16,
          20,
          24,
          28,
          32,
          36,
          40,
          44,
          48,
          52,
          56,
        ],
        "weights": Array [],
      }
    }
  >
    <IntlProvider
      defaultLocale="en"
      locale="en"
      messages={Object {}}
    >
      <BrowserRouter>
        <Switch>
          <DynamicRoute
            exact={true}
            loadAsyncModuleFn={[Function]}
            path="/**"
          />
        </Switch>
      </BrowserRouter>
    </IntlProvider>
  </ThemeProvider>
</Provider>
`);
  });

  test("routes", () => {
    expect(routes.homePage()).toMatchSnapshot();
  });
});
