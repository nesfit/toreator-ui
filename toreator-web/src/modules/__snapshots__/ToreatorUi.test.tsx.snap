// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToreatorUi/> routes 1`] = `
Object {
  "$$typeof": Symbol(react.lazy),
  "_ctor": [Function],
  "_result": null,
  "_status": -1,
}
`;
